contracts/ssnlist.scilla

```
@@ -268,34 +279,35 @@ transition drain_contract_balance (initiator : ByStr20)
   TransferFunds multisig_tag_addfunds bal initiator
 end
 
-(* @dev: Set the minstake of contract. Used by admin only. *)
+(* @dev: Set the staking parameters of contract. Used by admin only. *)
 (* @param min_stake: New minstake value *)
-(* @param initiator: The original caller who called the proxy *)
-transition update_minstake (min_stake : Uint128, initiator : ByStr20)
-  is_paused;
-  validate_proxy;
-  validate_admin initiator;
-  minstake := min_stake
-end
-
-(* @dev: Set the maxstake of contract. Used by admin only. *)
 (* @param max_stake: New maxstake value *)
+(* @param max_stake: New contract maxstake value *)
 (* @param initiator: The original caller who called the proxy *)
-transition update_maxstake (max_stake : Uint128, initiator : ByStr20)
+transition update_staking_parameter (min_stake : Uint128, max_stake : Uint128, contract_max_stake : Uint128, initiator : ByStr20)
   is_paused;
   validate_proxy;
   validate_admin initiator;
-  maxstake := max_stake
-end
+  less_than = builtin lt min_stake max_stake;
+  match less_than with
+  | True =>
+    less_than = builtin lt max_stake contract_max_stake;
+    match less_than with
+    | True =>
+      minstake := min_stake;
+      maxstake := max_stake;
+      contractmaxstake := contract_max_stake;
+      e = mk_update_staking_parameter_event initiator;
+      event e
+    | False => 
+      e = mk_update_staking_parameter_error initiator min_stake max_stake contract_max_stake;
+      event e
+    end
+  | False =>
+     e = mk_update_staking_parameter_error initiator min_stake max_stake contract_max_stake;
+     event e
```

should we throw an exception here?